brave/browser-laptop
# Add 'unauthorized' downloadItem state

When a user downloads a file to a location that they do not have access to, it currently just says "Downloading... 0%". This is to add a download state to indicate that the user doesn't have access to the director they are trying to download into.

Commit list:
- Add new download state
- Update state if directory is unauthorized
- Update electron download state change
- Add 'Unauthorized' download state to 'clear completed downloads' reducer
- Add isUnauthorized getter
- Add unauthorized state as part of condition for hiding progress display
- Add isUnauthorized to condition for rendering a download item
- Set unauthorized as a stop condition for downloads
- Set getL10nId for unauthorized downloads
- Add 'unauthorized' as a style class for finished download items
- Add downloadUnauthorized locale id for all languages
- Update tests
- Fix styling
- Change failed download wording
- Update download item logic to hide unneeded links
- Update tooltip to show localized download status

Fixes #7747

Submitter Checklist:

- [  ] Added/updated tests for this change (for new code or code which already has tests).

Test Plan:
1. Create a protected directory
2. Right click an Image in Brave and select Save As...
3. Attempt to save it in the protected directory
4. Download should stop and say "Failed - Unauthorized Access to Download Location"

Reviewer Checklist:

Tests
`npm run test -- --grep="downloadItem"`
